cmd/resource-query/resourcequery.go

```diff
@@ -0,0 +1,132 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"runtime"
+
+	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	kubectlcmd "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	kruntime "github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	utilyaml "github.com/GoogleCloudPlatform/kubernetes/pkg/util/yaml"
+)
+
+// TODO marekbiskup 2015-06-23: This is a temporary version, not meant for users.
+// Many things are missing:
+// * bash completion for files (json/yaml) - it partially exists but doesn't work
+// * handling files with multiple resources.
+// * reading a resource from stdin
+// * reading entire directory, also via http (like kubectl create)
+// * better error handling, better messages
+// * (?) handling of object kind and version (there is a version in the file,
+//   so we want to query it too.
+// * other template languages/output formats (like kubectl get)
+// * flags uhnified with kubectl get/create
+// * takes several files in one command - maybe it should take just one
+
+func tryDecodeSingleObj(data []byte) (object kruntime.Object, err error) {
+	// JSON is valid YAML, so this should work for everything.
+	json, err := utilyaml.ToJSON(data)
+	if err != nil {
+		return nil, err
+	}
+	obj, err := api.Scheme.Decode(json)
+	if err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+func getFileData(filename string) ([]byte, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return ioutil.ReadAll(file)
+}
+
+func extractFromFile(filename string) (object kruntime.Object, err error) {
+	data, err := getFileData(filename)
+	if err != nil {
+		return object, err
+	}
+
+	return tryDecodeSingleObj(data)
+}
+
+func NewResourceQueryCommand(in io.Reader, out, err io.Writer) *cobra.Command {
+	var filenames util.StringList
+	var template *string
+	cmd := &cobra.Command{
+		Use:   "resourcequery",
+		Short: "resourcequery is used to print a resource using a template.",
+		Long: `Print the resource defined in a file using a template.
+
+JSON and YAML input formats are accepted.
+This is an early alpha version of this command.
+Expect it to change or disappear.`,
+		Example: `// Print the resource name from the file.
+$ resourcequery -t {{.metadata.name}} -f pod.json`,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(kubectlcmd.ValidateArgs(cmd, args))
+
+			p, err := kubectl.NewTemplatePrinter([]byte(*template))
+			cmdutil.CheckErr(err)
+			for _, filename := range filenames {
+				obj, _ := extractFromFile(filename)
```

Comment:
Another possibility is not to parse the object, but to use json files directly. Then it would work for any json files, not only for ours. But I decided it's better that the tool gives an error in json files cannot be parsed.